misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt

```diff
@@ -36,12 +47,35 @@ class Keyspace(val sharded: Boolean) {
   fun shardCount() = if (sharded) 2 else 1
 }
 
-class VitessCluster(val config: DataSourceConfig, moshi: Moshi = Moshi.Builder().build()) {
-  val schemaDir = Paths.get(config.vitess_schema_dir)
+class VitessCluster(
+  val name: String,
+  resourceLoader: ResourceLoader,
+  val config: DataSourceConfig,
+  val moshi: Moshi = Moshi.Builder().build()
+) {
+  val schemaDir: Path
 
   init {
-    check(Files.isDirectory(schemaDir)) {
-      "can't find directory $schemaDir"
+    if (config.vitess_schema_dir != null) {
+      schemaDir = Paths.get(config.vitess_schema_dir)
+      StartVitessService.logger.warn {
+        "vitess_schema_dir is deprecated, use vitess_schema_resource_root instead"
+      }
+      check(Files.isDirectory(schemaDir)) {
+        "can't find directory $schemaDir"
+      }
+    } else {
+      val root = config.vitess_schema_resource_root
+          ?: throw IllegalStateException("vitess_schema_resource_root must be specified")
+      println("java.io.tmpdir = ${System.getProperty("java.io.tmpdir")}")
+      // We can't use Files::createTempDirectory because it creates a directory under the path
+      // /var/folders that is not possible to mount in Docker
+      schemaDir = Paths.get("/tmp/vitess_schema_${System.currentTimeMillis()}")
```

Comment:
if you’re trying to be unique, inject a TokenGenerator. currentTimeMillis() isn’t unique enough!

Reply:
Because VitessClusters are cached globally it's very awkward to inject into them. This will be sufficiently unique for our purposes I believe. Very rarely does a developer start more than a single cluster at exactly the same millisecond.